app/services/sets/product_set.rb

```diff
@@ -153,5 +153,12 @@ def notify_bugsnag(error, product, variant, variant_attributes)
         report.add_metadata(:variant_error, variant.errors.first) unless variant.valid?
       end
     end
+
+    def create_stock_for_variant(variant, on_demand, on_hand)
+      variant.on_demand = on_demand if on_demand.present?
+      variant.on_demand = variant.on_demand_desired if variant.on_demand_desired.present?
+      variant.on_hand = on_hand.to_i if on_hand.present?
+      variant.on_hand = variant.on_hand_desired.to_i if variant.on_hand_desired.present?
```

Comment:
It's not clear to me which value is used in which case here. Looking only at this logic, the stock could even be set twice when updated. So first it would set the old on_hand value when it's present and then set the new desired value. This is a bit inefficient but also confusing to follow.

I think that it would be clearer to have one branch of logic:

* Form always sets the desired value.
* We always use the desired value to update stock.

Reply:
Yes, it was a bit cluttered. After a few trials (and errors), I think this is better now.
The conditional I added is because in legacy mode, you enter the create variant with on Hand/on demand whereas in v3 you do not.